### PR TITLE
[Transforms] elide exports with no value side

### LINKED ISSUE #8037: [Transforms] Exports for non-value imports not elided in ES6 modules
**Expected**:

```js
//// [server.ts]

class c {
}
interface i {
}
module m {
    export var x = 10;
}
var x = 10;
module uninstantiated {
}
export { c };
export { c as c2 };
export { i, m as instantiatedModule };
export { uninstantiated };
export { x };

//// [server.js]
class c {
}
var m;
(function (m) {
    m.x = 10;
})(m || (m = {}));
var x = 10;
export { c };
export { c as c2 };
export { m as instantiatedModule };
export { x };
```

**Actual**:

```js
//// [server.ts]

class c {
}
interface i {
}
module m {
    export var x = 10;
}
var x = 10;
module uninstantiated {
}
export { c };
export { c as c2 };
export { i, m as instantiatedModule };
export { uninstantiated };
export { x };

//// [server.js]
class c {
}
var m;
(function (m) {
    m.x = 10;
})(m || (m = {}));
var x = 10;
export { c };
export { c as c2 };
export { i, m as instantiatedModule };
export { uninstantiated };
export { x };
```

Failed tests:
*  es6ExportClause.js

### PR DESCRIPTION
**Pull request by vladima**
fixes #8037

### CODE REVIEW

**Comment by sandersn** [comment]
:+1: although @rbuckton should also take a look.

File: src/compiler/transformers/module/es6.ts [hunk 1]
```diff
@@ -34,19 +34,53 @@ namespace ts {
...
             if (node.isExportEquals) {
                 return undefined; // do not emit export equals for ES6
             }
             const original = getOriginalNode(node);
             return nodeIsSynthesized(original) || resolver.isValueAliasDeclaration(original) ? node: undefined;
         }
 
+        function visitExportDeclaration(node: ExportDeclaration): ExportDeclaration {
+            if (!node.exportClause) {
+                return node; // export * is always emitted
```
Comment (mhegazy) [inline src/compiler/transformers/module/es6.ts]: mmm. should not we check that the module we are exporting has some concrete value to export.. i.e.  `resolver.moduleExportsSomeValue(...)`

Comment (vladima) [inline src/compiler/transformers/module/es6.ts]: @mhegazy  I think we should though in the old emitter we are doing this only for downlevel compilation that does not have native support for `export *`

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**